.changeset/wild-eyes-grab.md

```diff
@@ -0,0 +1,10 @@
+---
+'@backstage/plugin-search-backend-module-elasticsearch': minor
+---
+
+**BREAKING** The ElasticSearch indexer will now delete stale indices matching the indexer's pattern.
+
+An indexer using the `some-type-index__*` pattern will remove indices matching this pattern after indexation
```

Comment:
Is there an action that the adopter should take before upgrading to this new version? For example, updating their index patterns 🤔

Reply:
I've added an action to verify that the pattern isn't conflicting with other indices not managed by Backstage
(which should already be an "issue" since the indexer removes aliases on these indices)